File: judge/src/tested/languages/templates/haskell/SpecificEvaluationUtils.hs
{-# LANGUAGE OverloadedStrings #-}
module SpecificEvaluationUtils where

import Values

-- | Report the result of an evaluation to the judge. This method should only
-- be called once, otherwise things will break.
-- Note that the first parameter, the file path, will be provided and must not
-- be changed. It should be passed to `evaluated` as is.
evaluated :: FilePath -- ^ Where the result should be written.
          -> Bool     -- ^ The result of the evaluation.
          -> String   -- ^ A string version of the actual value. Optional.
          -> String   -- ^ A string version of the expected value. Optional.
          -> [String] -- ^ A list of messages for the user.
          -> IO ()
evaluated = sendSpecificEvaluated

File: judge/src/tested/languages/python/templates/specific_evaluation_utils.py
import values

from typing import List, Optional


out = None


def evaluated(result: bool,
              readable_expected: Optional[str] = None,
              readable_actual: Optional[str] = None,
              messages: Optional[List[str]] = None):
    """
    Report the result of an evaluation to the judge. This method should only
    be called once per evaluation. Calling this multiple times will result in
    undefined behaviour.

    :param result: The result of the evaluation (True or False).

    :param readable_expected: A string version of the channel value. Optional;
                              if not given, the judge will produce one on a
                              best-efforts basis.
    :param readable_actual: A string version of the actual value. Optional; if
                            not given, the judge will produce one on a best-
                            efforts basis.
    :param messages: Optional list of messages to be shown to the student.
    """
    if messages is None:
        messages = []

    values.send_evaluated(out,
                          result, readable_expected, readable_actual, messages)
